### PR TITLE
add model and exceptions

### PR DESCRIPTION
**Pull request by mdayican**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/BPS-765

Change description

adding the queue model and exceptions
 
Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentMessage.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public final List<PaymentInfo> payments;
+
+
+    public PaymentMessage(
+        @JsonProperty(value = "envelope_id", required = true) String envelopeId,
+        @JsonProperty(value = "ccd_case_number", required = true) String ccdCaseNumber,
+        @JsonProperty(value = "is_exception_record", required = true) boolean isExceptionRecord,
+        @JsonProperty(value = "po_box", required = true) String poBox,
+        @JsonProperty(value = "jurisdiction", required = true) String jurisdiction,
+        @JsonProperty(value = "payments", required = true) List<PaymentInfo> payments) {
```
Comment (aliveni) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentMessage.java]: is `site_id` missing? it's there in the jira ticket

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentMessage.java]: we will map jurisdiction  to site id .there is a site SiteConfiguration class which reads these stite id comnfigs.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentInfo.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class PaymentInfo {
+
+    public final String documentControlNumbers;
+
+    public PaymentInfo(@JsonProperty(value = "document_control_numbers", required = true)
```
Comment (aliveni) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentInfo.java]: I think the property name should be `document_control_number`?

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentInfo.java]: fixed

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentInfo.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class PaymentInfo {
+
+    public final String documentControlNumber;
+
+    public PaymentInfo(@JsonProperty(value = "document_control_number", required = true)
+                           String documentControlNumber) {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentInfo.java]: Hm, this looks incorrectly intended

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/handler/MessageProcessingResultType.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler;
+
+public enum MessageProcessingResultType {
+    SUCCESS,
+    UNRECOVERABLE_FAILURE,
+    POTENTIALLY_RECOVERABLE_FAILURE
+}
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/handler/MessageProcessingResultType.java]: This looks like a copy/paste from another project. Perhaps we won't even need similar approach, perhaps development can be done individually in some way. Especially when idempotency is guaranteed from day one unlike the other project

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentMessage.java (hunk 1 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentMessage.java]: You should be mapping pobox or container to siteid and not jurisdiction

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/model/PaymentMessage.java]: I will use po box map to site_id.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/handler/MessageProcessingResultType.java (hunk 4 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/handler/MessageProcessingResultType.java]: for now 
SUCCESS . = success no retry , http status 200,201
UNRECOVERABLE_FAILURE = failure no retry, http 409
POTENTIALLY_RECOVERABLE_FAILURE  = failure retry ,other http status codes

**Pull request merged**